### DISCUSSION TITLE
How to programmatically make the form components touched .

### DISCUSSION THREAD

**Discussion by adityapatil9191** [data-driven-forms/react-forms#1461] [Need help]
Hello,
I am working on creation of the form components. I want to programmatically change their state to touched. How can I do that? @rvsia

**Comment by adityapatil9191** [comment]
Please find the link for reference https://codesandbox.io/p/sandbox/react-data-driven-autocomplete-forked-9tkfvt?file=%2Fsrc%2FForm.jsx%3A17%2C26. When I move from one accordion to another, I want to highlight validations for that accordion when I move away from it.

**Comment by rvsia** [comment]
You can find all the input fields in the children and do `focus`/`blur`.

**Reply by adityapatil9191** [reply]
could you please help me any example ? This is what I am trying to do 
const formOptions = useFormApi();
const {lastOpened, setFormState} = useAppContext();
const {fields} = useFieldsApi(props);
//This if condition is to extract all the fields which were last opened and it is working as expected.
if (fields && fields.length) {
   const filteredFieldItems = fields.filter((field:any)=> field.name === lastOpened)
}
if(filteredFieldItems && filtteredFieldItems.length) {
   filteredFieldItems[0].fields.forEach((field:any)=> {
   **formOptions.focus(field.name);**
})
} 
The line highlighted above does not focus the fields. Is that a correct syntax? I am able to extract the fields . @rvsia

**Comment by adityapatil9191** [comment]
@rvsia please let me know if you had a chance to review the above code

**Comment by adityapatil9191** [comment]
Hello can anyone please help me with the status of the above request?
